### PR TITLE
Optimize the layer_norm operator with AVX intrinsic function

### PR DESCRIPTION
**Pull request by yihuaxu**
According to the performance status of Layer Norm, just implemented the intrinsic function's optimization to accelerate the data processing.

**Platform**: Intel(R) Xeon(R) Gold 6151 CPU @ 3.00GHz / Intel(R) Xeon(R) CPU E5-2699 v3 @ 2.30GHz
**Batch Size**: 1
**Command**: build/paddle/fluid/inference/tests/api/test_analyzer_dam --infer_model=PaddlePaddle/pretrained_models/dam --infer_data=third_party/inference_demo/dam/data.txt --gtest_filter=Analyzer_dam.profile --batch_size=1 --repeat=1 --test_all_data=true --num_threads=1 --use_analysis=false 

The following is the comparison with the different scenarios.
image

### CODE REVIEW

**Comment by yihuaxu** [comment]
@tensor-tang @luotao1 Please help us review this PR

**Review by tensor-tang — APPROVED** [review]
Thanks for your contribution @yihuaxu 

The jit_kernel is migrating to jit code, and we are using jitcode to generate the runtime code on different platform, instead of using intrinsic now. 

We will let you know when it's ready.

File: paddle/fluid/operators/CMakeLists.txt [hunk 1]
```diff
@@ -315,11 +315,18 @@ op_library(fusion_lstm_op DEPS jit_kernel)
...
+        op_library(layer_norm_op DEPS cub jit_kernel)
+    else(NOT WIN32)
+        op_library(layer_norm_op DEPS cub)
+    endif(NOT WIN32)
     op_library(reduce_mean_op DEPS cub)
     op_library(affine_channel_op DEPS cub)
 else()
     op_library(conv_op DEPS vol2col im2col)
+    if (NOT WIN32)
+        op_library(layer_norm_op DEPS jit_kernel)
```
Comment (tensor-tang) [inline paddle/fluid/operators/CMakeLists.txt]: As for windows part @dzhwinter @wopeizl  is this ok to you?

**Review by tensor-tang — CHANGES_REQUESTED** [review]
@yihuaxu 我突然意识到这部分code在Mac也需要被注释掉吧，我们线下也会测试Mac上的单测。

File: paddle/fluid/operators/CMakeLists.txt (hunk 1 above)
Comment (tensor-tang) [reply paddle/fluid/operators/CMakeLists.txt]: 这一部分与@dzhwinter @wopeizl 沟通后是没什么问题。

**Review by yihuaxu — COMMENTED** [review]
Remove the remnant code

**Review by tensor-tang — APPROVED** [review]

**Pull request merged**